service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapper.java

```diff
@@ -155,19 +157,26 @@ private String prepareReportLevelNarrativeStatements(DiagnosticReport diagnostic
             String comment = PREPENDED_TEXT_FOR_CODED_DIAGNOSIS + codedDiagnosisText;
 
             String narrativeStatementFromCodedDiagnosis = buildNarrativeStatementForDiagnosticReport(
-                diagnosticReport, comment
+                diagnosticReport, LABORATORY_RESULT_COMMENT.getCode(), comment
             );
 
             reportLevelNarrativeStatements.append(narrativeStatementFromCodedDiagnosis);
         }
 
+        if (reportLevelNarrativeStatements.length() == 0 && !diagnosticReport.hasResult()) {
+            String narrativeStatementFromCodedDiagnosis = buildNarrativeStatementForDiagnosticReport(
```

Comment:
From looking at the mapping spreadsheet, the dummy/default observation should look like this
```
<component typeCode="COMP" contextConductionInd="true">
    <NarrativeStatement classCode="OBS" moodCode="EVN">
        <id root="5E496953-065B-41F2-9577-BE8F2FBD0757"/>
        <text mediaType="text/x-h7uk-pmip">CommentType:AGGREGATE COMMENT SET
            CommentDate:20100225154100

            EMPTY REPORT</text>
        <statusCode code="COMPLETE"/>
        <availabilityTime value="20100225154100"/>
    </NarrativeStatement>
</component>
```

So I think `"DUMMY"` should be `CommentType.AGGREGATE_COMMENT_SET.getCode()` and `EMPTY` should be `"EMPTY REPORT"`. Also makes it clearer this is a dummy/generated narrative statement

Reply:
Also I'd put this logic in some private method called `buildDefaultNarrativeStatementForMissingObservation` or something like that

Reply:
done